airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/IcebergDestination.java

```diff
@@ -0,0 +1,90 @@
+import io.airbyte.integrations.destination.iceberg.config.catalog.IcebergCatalogConfigFactory;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.SparkSession.Builder;
+
+@Slf4j
+public class IcebergDestination extends BaseConnector implements Destination {
+
+    private final IcebergCatalogConfigFactory icebergCatalogConfigFactory;
+
+    public IcebergDestination() {
+        this.icebergCatalogConfigFactory = new IcebergCatalogConfigFactory();
+    }
+
+    @VisibleForTesting
+    public IcebergDestination(IcebergCatalogConfigFactory icebergCatalogConfigFactory) {
+        this.icebergCatalogConfigFactory = Objects.requireNonNullElseGet(icebergCatalogConfigFactory,
+            IcebergCatalogConfigFactory::new);
+    }
+
+    public static void main(String[] args) throws Exception {
+        new IntegrationRunner(new IcebergDestination()).run(args);
+    }
+
+    @Override
+    public AirbyteConnectionStatus check(JsonNode config) {
+        try {
+            IcebergCatalogConfig icebergCatalogConfig = icebergCatalogConfigFactory.fromJsonNodeConfig(config);
+            icebergCatalogConfig.check();
+
+            //getting here means Iceberg catalog check success
+            return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
+        } catch (final Exception e) {
+            log.error("Exception attempting to access the Iceberg catalog: ", e);
+            Throwable rootCause = getRootCause(e);
+            String errMessage =
+                "Could not connect to the Iceberg catalog with the provided configuration. \n" + e.getMessage()
+                + ", root cause: " + rootCause.getClass().getSimpleName() + "(" + rootCause.getMessage() + ")";
+            return new AirbyteConnectionStatus()
+                .withStatus(AirbyteConnectionStatus.Status.FAILED)
+                .withMessage(errMessage);
+        }
+    }
+
+    private Throwable getRootCause(Throwable exp) {
+        Throwable curCause = exp.getCause();
+        if (curCause == null) {
+            return exp;
+        } else {
+            return getRootCause(curCause);
+        }
+    }
+
+    @Override
+    public AirbyteMessageConsumer getConsumer(JsonNode config,
+        ConfiguredAirbyteCatalog catalog,
+        Consumer<AirbyteMessage> outputRecordCollector) {
+        final IcebergCatalogConfig icebergCatalogConfig = this.icebergCatalogConfigFactory.fromJsonNodeConfig(config);
+        Map<String, String> sparkConfMap = icebergCatalogConfig.sparkConfigMap();
+
+        Builder sparkBuilder = SparkSession.builder()
+            .master("local")
```

Comment:
does local mean that apache spark will run in the same container as the connector itself? does this mean that you cant target an already existing cluster in your dc? Is this applicable for the Netflix usecase https://github.com/airbytehq/airbyte/issues/2833?

Reply:
yes, in spark's local mode, driver and executor runs on the same container as the connector itself.  
it can also runs on an already existing cluster, but i don't think it's necessary, because of :

1. all incoming data are from the connector's container, it's the bottleneck of the whole syncing job; data are not distributed, so running spark job on existing cluster does not benefit from spark cluster.
2. submitting spark job to an already existing cluster may cost extra time, for waiting spark worker resource, and extra network I/O, etc.,

Reply:
Yeah, I get your point :+1: